.github/workflows/benchmark.yml

```diff
@@ -1,51 +1,129 @@
-name: Benchmark PR vs. release-0.x
+name: Benchmark PR vs main
 
 on:
+  workflow_dispatch:
   pull_request:
     branches:
-      - release-0.x
+      - main
+    paths:
+      - '**.swift'
+      - '**.yml'
 
 concurrency:
   group: ${{ github.workflow }}-${{ github.ref }}-benchmark
   cancel-in-progress: true
 
+env:
+  OTEL_ENABLE_BENCHMARKS: true
+
 jobs:
   benchmark:
-    name: Check benchmark regressions
-    runs-on: ubuntu-24.04
+    name: Check benchmark regressions (Swift ${{ matrix.swift.version }})
+    runs-on: ubuntu-latest
     timeout-minutes: 15
     continue-on-error: true
+    permissions:
+      issues: write
+      pull-requests: write
+      contents: read
+
+    strategy:
+      fail-fast: false
+      matrix:
+        swift:
+          - version: "6.1"
+            image: swift:6.1
+
+    container:
+      image: ${{ matrix.swift.image }}
+      options: --dns 1.1.1.1 --dns 8.8.8.8
 
     steps:
-      - name: Checkout
-        uses: actions/checkout@v5.0.0
+      - uses: actions/checkout@v4
```

Comment:
Should we stay on v5? If not, Dependabot will just update it next week.

Reply:
Sounds good. I'm gonna remove the CI from this one but will include this in the PR that adds it back in.

Reply:
Actually, I'll do it now and just mark the job as skipped for now.

Reply:
Done in https://github.com/swift-otel/swift-otel/pull/345/commits/a83fe8b59076ec4628916de0bc3b392501abc675